odl/util/ufuncs.py

```diff
@@ -335,10 +334,15 @@ def wrapper(self, out1=None, out2=None):
     elif n_in == 2:
         if n_out == 1:
             def wrapper(self, x2, out=None):
-                try:
+                if x2 in self.vector.space:
                     x2 = x2.ntuple
+
+                try:
+                    # Try to reshape to linear data
+                    x2 = x2.reshape(self.vector.size,
+                                    order=self.vector.space.order)
                 except AttributeError:
```

Comment:
Hm, this is for `NtuplesBaseVector`. Does it at all make sense to flatten? Shouldn't it be an error if something multi-dimensional is given? Numpy bails out in that case since broadcasting is not possible.